src/itwinai/torch/trainer.py

```diff
@@ -1220,3 +1231,382 @@ def dist_train(
+        self._set_strategy_and_init_ray(strategy)
+        self._set_configs(config=config)
+
+    def _set_strategy_and_init_ray(self, strategy: str):
+        """Set the distributed training strategy. This will initialize the ray backend.
+
+        Args:
+            strategy (str): The strategy to use for distributed training.
+                Must be one of ["ddp", "deepspeed"].
+
+        Raises:
+            ValueError: If an unsupported strategy is provided.
+        """
+        if strategy == "ddp":
+            self.strategy = RayDDPStrategy()
+        elif strategy == "deepspeed":
+            self.strategy = RayDeepSpeedStrategy()
+        else:
+            raise ValueError(f"Unsupported strategy: {strategy}")
+
+    def _set_configs(self, config: Dict):
+        self.config = deep_update(DEFAULT_CONFIG, config)
+        self._set_scaling_config()
+        self._set_tune_config()
+        self._set_run_config()
+        self._set_train_loop_config()
+
+    @property
+    def device(self) -> str:
+        """Get the current device from distributed strategy.
+        Returns:
+            str: Device string (e.g., "cuda:0").
+        """
+        return self.strategy.device()
+
+    def create_dataloaders(
+        self,
+        train_dataset: Dataset,
+        validation_dataset: Dataset | None = None,
+        test_dataset: Dataset | None = None,
+        batch_size: int = 1,
+        shuffle_train: bool | None = False,
+        shuffle_test: bool | None = False,
+        shuffle_validation: bool | None = False,
+        sampler: Union[Sampler, Iterable, None] = None,
+        collate_fn: Callable[[List], Any] | None = None,
+    ) -> None:
+        """Create data loaders for training, validation, and testing.
+
+        Args:
+            train_dataset (Dataset): The training dataset.
+            validation_dataset (Dataset, optional): The validation dataset. Defaults to None.
+            test_dataset (Dataset, optional): The test dataset. Defaults to None.
+            batch_size (int, optional): Batch size for data loaders. Defaults to 1.
+            shuffle_train (bool, optional): Whether to shuffle the training dataset.
+                Defaults to False.
+            shuffle_test (bool, optional): Whether to shuffle the test dataset.
+                Defaults to False.
+            shuffle_validation (bool, optional): Whether to shuffle the validation dataset.
+                Defaults to False.
+            sampler (Union[Sampler, Iterable, None], optional): Sampler for the datasets.
+                Defaults to None.
+            collate_fn (Callable[[List], Any], optional):
+                Function to collate data samples into batches. Defaults to None.
+        """
+        self.train_dataloader = self.strategy.create_dataloader(
+            dataset=train_dataset,
+            batch_size=batch_size,
+            shuffle=shuffle_train,
+            sampler=sampler,
+            collate_fn=collate_fn,
+        )
```

Comment:
why the torch_rng is not passed to the dataloader? 
Why does this diverge from the implementation in the TorchTrainer (see below)?

```python
self.train_dataloader = self.strategy.create_dataloader(
            dataset=train_dataset,
            batch_size=self.config.batch_size,
            num_workers=self.config.num_workers_dataloader,
            pin_memory=self.config.pin_gpu_memory,
            generator=self.torch_rng,
            shuffle=self.config.shuffle_train,
        )
```

Same for the code blocks below

Reply:
I kept this more minimal for now because I had not made a final design choice about the config, and the dictionary implementation so far does not support default values. As this will change anyway, is it okay if I just make a note of this to change once I figure out the config situation?

Reply:
Okay I decided to make it more in line with the TorchTrainer implementation anyway, even if we change it later